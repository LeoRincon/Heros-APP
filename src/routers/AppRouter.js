import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

import { LoginScreen } from '../components/login';
import { DashboardRouter } from './';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <div>
        <Routes>
          <Route path='/*' element={<DashboardRouter />} />
          <Route exact path='/login' element={<LoginScreen />} />
        </Routes>
      </div>
    </BrowserRouter>
  );
};
