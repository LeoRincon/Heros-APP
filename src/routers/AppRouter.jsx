import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

import { LoginScreen } from '../components/login';
import { DashboardRouter } from '.';
import { PrivateRoute } from '.';
import { PublicRoute } from '.';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route
          path='/login'
          element={
            <PublicRoute>
              <LoginScreen />
            </PublicRoute>
          }
        />
        <Route
          path='/*'
          element={
            <PrivateRoute>
              <DashboardRouter />
            </PrivateRoute>
          }
        />
      </Routes>
    </BrowserRouter>
  );
};
