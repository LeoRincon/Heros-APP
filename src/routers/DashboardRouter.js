import React from 'react';
import { Navbar } from '../components/ui/NavBar';
import { Routes, Route } from 'react-router-dom';

import { MarvelScreen } from '../pages/marvel';
import { DcScreen } from '../pages/dcScreen';
import { HeroScreen } from '../components/heroes';

export const DashboardRouter = () => {
  return (
    <>
      <Navbar />
      <div className='container mt-3'>
        <Routes>
          <Route exact path='/marvel' element={<MarvelScreen />} />
          <Route exact path='/heroe/:heroeId' element={<HeroScreen />} />
          <Route exact path='/dc' element={<DcScreen />} />
          <Route path='*' element={<MarvelScreen />} />
        </Routes>
      </div>
    </>
  );
};
