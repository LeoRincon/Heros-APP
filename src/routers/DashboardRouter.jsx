import React from 'react';
import { Navbar } from '../components/ui/NavBar';
import { Routes, Route } from 'react-router-dom';

import { MarvelScreen } from '../pages/marvel';
import { DcScreen } from '../pages/dcScreen';
import { HeroScreen } from '../components/heroes';
import { SearchScreen } from '../pages/search/SearchScreen';

export const DashboardRouter = () => {
  return (
    <>
      <Navbar />
      <div className='container mt-3'>
        <Routes>
          <Route exact path='/marvel' element={<MarvelScreen />} />
          <Route exact path='/hero/:heroeId' element={<HeroScreen />} />
          <Route exact path='/dc' element={<DcScreen />} />
          <Route exact path='/search' element={<SearchScreen />} />
          <Route path='*' element={<MarvelScreen />} />
        </Routes>
      </div>
    </>
  );
};
