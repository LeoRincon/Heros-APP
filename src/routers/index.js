export { DashboardRouter } from './DashboardRouter';
export { PrivateRoute } from './PrivateRoute';
export { AppRouter } from './AppRouter';
export { PublicRoute } from './PublicRoute';
