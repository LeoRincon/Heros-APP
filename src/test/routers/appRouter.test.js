import { mount } from 'enzyme';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('when <appRouter> is called', () => {
  test('should show the login if it is not authenticated', () => {
    const contextValue = {
      user: {
        logged: false,
      },
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    console.log(wrapper);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h2').text().trim()).toBe('Login');
  });

  test('should render the dashboard when the user is authenticated', () => {
    const contextValue = {
      user: {
        logged: true,
        name: 'Lenardo',
      },
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('navbar').exists).toBeTruthy();
  });
});
