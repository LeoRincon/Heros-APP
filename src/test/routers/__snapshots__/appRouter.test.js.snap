// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when <appRouter> is called should render the dashboard when the user is authenticated 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end px-4"
    >
      <nav
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Lenardo
        </span>
        <button
          className="nav-item nav-link btn"
          exact="true"
          onClick={[Function]}
          to="/login"
        >
          Logout
        </button>
      </nav>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <div
        className="row animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Spider Man"
                className="card-img"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                   (
                  Peter Parker
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Captain America"
                className="card-img"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Captain America
                   (
                  Steve Rogers
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Iron Man"
                className="card-img"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Iron Man
                   (
                  Tony Stark
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Thor"
                className="card-img"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Thor
                   (
                  Thor Odinson
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hulk"
                className="card-img"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hulk
                   (
                  Bruce Banner
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Wolverine"
                className="card-img"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Wolverine
                   (
                  James Howlett
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Daredevil"
                className="card-img"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Daredevil
                   (
                  Matthew Michael Murdock
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Daredevil #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hawkeye"
                className="card-img"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hawkeye
                   (
                  Clinton Francis Barton
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Cyclops"
                className="card-img"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Cyclops
                   (
                  Scott Summers
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    X-Men #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Silver Surfer"
                className="card-img"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Silver Surfer
                   (
                  Norrin Radd
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`when <appRouter> is called should show the login if it is not authenticated 1`] = `
<div
  className="container mt-5"
>
  <h2>
    Login
  </h2>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
